File: src/main/java/dev/slne/discord/guild/DiscordGuild.java
package dev.slne.discord.guild;

public class DiscordGuild {

    private String guildId;
    private String categoryId;
    private String discordTranscriptId;
    private String serverTranscriptId;

    public DiscordGuild(String guildId, String categoryId, String discordTranscriptId, String serverTranscriptId) {
        this.guildId = guildId;
        this.categoryId = categoryId;
        this.discordTranscriptId = discordTranscriptId;
        this.serverTranscriptId = serverTranscriptId;
    }

    public String getGuildId() {
        return guildId;
    }

    public String getCategoryId() {
        return categoryId;
    }

    public String getDiscordTranscriptId() {
        return discordTranscriptId;
    }

    public String getServerTranscriptId() {
        return serverTranscriptId;
    }

}

File: src/main/kotlin/dev/slne/discord/guild/DiscordGuild.kt
package dev.slne.discord.guild

import dev.slne.discord.getBean
import dev.slne.discord.guild.role.DiscordRolePermissions
import dev.slne.discord.ticket.TicketType
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.entities.Role

data class DiscordGuild(
    val guildId: String,
    val whitelistRoleId: String,

    val roles: List<DiscordRolePermissions>,
    val ticketChannels: Map<TicketType, String>
) {
    val whitelistRole: Role?
        get() = getBean<JDA>().getGuildById(guildId)?.getRoleById(whitelistRoleId)
}